### ISSUE TITLE
get: cannot get local files that haven't been pushed when `dvc remote default <remote>` has been set

### ISSUE THREAD

**Issue by ciaochiaociao** [iterative/dvc#5967]
# Bug Report

## Description

After using `dvc remote default <remote>` to set a default remote, `dvc get` can not retrieve the files which are only in local cache (because they haven't been pushed). It should have retrieved files either from local or remote. 

Workaround that I have tried and succeeds: unsetting the default remote

### Reproduce

1. git init
2. dvc init
3. echo 'This is a dvc get test file' > dvc_get.test
4. dvc add dvc_get.test
5. git add .
6. git commit -m 'add a file'
7. dvc get -o /tmp/file1 . dvc_get.test  # success. get files from .dvc/.cache
8. dvc remote default <remote>
9. dvc get -o /tmp/file2 . dvc_get.test  # failed
```
ERROR: unexpected error

Having any troubles? Hit us up at https://dvc.org/support, we are always happy to help!
```

### Expected

It should get us files either from local or remote

### Environment information
**Output of `dvc doctor`:**

```console
$ dvc doctor
DVC version: 2.1.0 (conda)
---------------------------------
Platform: Python 3.9.2 on Linux-4.4.0-140-generic-x86_64-with-glibc2.23
Supports: http, https, ssh
```

**Additional Information (if any):**
**workaroud**:
1. unsetting default remote by `dvc remote default --unset`
2. then run `dvc get -o /tmp/file2 . dvc_get.test` again => success

**Comment by karajan1001** [comment]
@ciaochiaociao Thank you for your report.

**Comment by dberenbaum** [comment]
@ciaochiaociao What is your use case? `dvc get` is for downloading files from other projects, so it's not a usual case to want to do it locally within your project. If you want to retrieve files from within your project, you would normally use `dvc checkout`.

**Comment by karajan1001** [comment]
@dberenbaum, the result won't be changed in other cases, I think. It is only a test script, so set the remote repo and downloading repo to the same one for convenience.

![image](https://user-images.githubusercontent.com/6745454/117224606-f7005200-ae42-11eb-9dfd-d158e6d9968e.png)

**Comment by ciaochiaociao** [comment]
> @ciaochiaociao What is your use case? `dvc get` is for downloading files from other projects, so it's not a usual case to want to do it locally within your project. If you want to retrieve files from within your project, you would normally use `dvc checkout`.

I have tensorboard run directory as an one of `outs` in my stage `train`. And to compare each run, I have to put all runs into a folder. So that's why I want to use `dvc get` to download from my local projects. 

Note: I know that usually all runs are put into a single folder when conducting different experiments. But the way I build my project with dvc is following dvc's philosophy of **each version model/data and other related files to this version should be tracked by `git`**. So that is why I decided to track tb run folder by dvc. This gets me rid of naming my tensorboard run, and gives the power of git to reference each run to its corresponding commit in git. And that's also why I need the above workaround to download different runs by `dvc get` to a single folder for comparison.

**Comment by dberenbaum** [comment]
My understanding was that `dvc get` will pull the data from the default DVC remote. Here's the relevant docs section from https://dvc.org/doc/command-reference/get:

> ⚠️ DVC repos should have a default DVC remote containing the target actual for this command to work. The only exception is for local repos, where DVC will try to copy the data from its cache first.

It does say that a local repo will try to copy the data from its cache first, so it makes sense to mark it as a bug.

@ciaochiaociao Let me see if I understand:

1. You start each experiment with an empty tensorboard run directory.
2. You run the experiment and track the directory's output and commit to Git.
3. You now want to compare by populating the directory with the results from multiple experiments.

Is that correct?

**Comment by ciaochiaociao** [comment]
> My understanding was that `dvc get` will pull the data from the default DVC remote. Here's the relevant docs section from https://dvc.org/doc/command-reference/get:
> 
> > ⚠️ DVC repos should have a default DVC remote containing the target actual for this command to work. The only exception is for local repos, where DVC will try to copy the data from its cache first.
> 
> It does say that a local repo will try to copy the data from its cache first, so it makes sense to mark it as a bug.
> 
> @ciaochiaociao Let me see if I understand:
> 
> 1. You start each experiment with an empty tensorboard run directory.
> 2. You run the experiment and track the directory's output and commit to Git.
> 3. You now want to compare by populating the directory with the results from multiple experiments.
> 
> Is that correct?

Correct

**Comment by jorgeorpinel** [comment]
May be the same issue as https://github.com/iterative/dvc/issues/4874#issuecomment-725528728.

**Comment by dberenbaum** [comment]
This appears to be fixed for me in 3.13.3.
